File: picture-server/src/main/java/com/applory/pictureserver/domain/user/UserRepositoryCustom.java
package com.applory.pictureserver.domain.user;

import com.applory.pictureserver.domain.user.querydto.SellerListVM;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Repository;

@Repository
public interface UserRepositoryCustom {
    Page<User> findClientUserBySearch(UserDto.SearchClient search, Pageable pageable);

    Page<SellerListVM> findSellerUserBySearch(UserDto.SearchSeller search, Pageable pageable);
}

File: picture-server-kt/src/main/kotlin/com/applory/pictureserverkt/user/UserRepositoryCustom.kt
package com.applory.pictureserverkt.user

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.stereotype.Repository

@Repository
interface UserRepositoryCustom {
    fun findClientUserBySearch(search: UserDto.SearchClient, pageable: Pageable): Page<User>

    fun findSellerUserBySearch(search: UserDto.SearchSeller, pageable: Pageable): Page<User>
}
